app/src/main/java/com/github/se/project/ui/components/LessonEditor.kt

```diff
@@ -267,21 +296,30 @@ fun validateLessonInput(
     selectedSubject: MutableState<Subject>,
     selectedLanguages: List<Language>,
     date: String,
-    time: String
+    time: String,
+    latitude: Double,
+    longitude: Double
 ): String? {
-  for (entry in
+  val requiredFields =
       mapOf(
-              "title" to title,
-              "description" to description,
-              "subject" to selectedSubject.value.name,
-              "language" to selectedLanguages.joinToString { it.name },
-              "date" to date,
-              "time" to time,
-          )
-          .entries) {
-    if (entry.value.isEmpty()) {
-      return "${entry.key} is missing"
+          "title" to title,
+          "description" to description,
+          "subject" to selectedSubject.value.name,
+          "language" to selectedLanguages.joinToString { it.name },
+          "date" to date,
+          "time" to time)
+
+  // Check if any required field is empty
+  for ((field, value) in requiredFields) {
+    if (value.isEmpty()) {
+      return "$field is missing"
     }
   }
-  return null
+
+  // Check if location has been set
+  if (latitude == 0.0 && longitude == 0.0) {
```

Comment:
I think it would be cleaner to do the check for the location in the same way as previous checks, by adding maybe a new (field,value) pair to the requiredField, ie. "location" to if (latitude == 0.0 || longitude == 0.0) "" else "set"